mlir/python/CMakeLists.txt

```diff
@@ -423,7 +423,30 @@ declare_mlir_python_extension(MLIRPythonExtension.Core
     MLIRCAPIInterfaces
 
     # Dialects
+    MLIRCAPIAMDGPU
+    MLIRCAPIArith
+    MLIRCAPIAsync
+    MLIRCAPIControlFlow
     MLIRCAPIFunc
+    MLIRCAPIGPU
+    MLIRCAPILLVM
+    MLIRCAPILinalg
+    MLIRCAPIMLProgram
+    MLIRCAPIMath
+    MLIRCAPIMemRef
+    MLIRCAPINVGPU
+    MLIRCAPINVVM
+    MLIRCAPIOpenMP
+    MLIRCAPIPDL
+    MLIRCAPIQuant
+    MLIRCAPIROCDL
+    MLIRCAPISCF
+    MLIRCAPIShape
+    MLIRCAPISparseTensor
+    MLIRCAPITensor
+    MLIRCAPITransformDialect
+    MLIRCAPIVector
+    MLIRCAPIRemainingDialects
```

Comment:
I don't think we want "Core" bindings to depend on all dialects...

Reply:
Yeah. I need to activate some old caches for this, but I think it was this issue that kept me from doing this the first time around.

If we can get the dependency dag such that `RegisterEverything` carries the weight of "everything" but is strictly optional, I think that is what we are going for.